packages/@aws-cdk/region-info/lib/user-data.ts

```diff
@@ -0,0 +1,95 @@
+import fs = require('fs');
+import jsonschema = require('jsonschema');
+import { IRegionInfo } from './iregion-info';
+import builtIns = require('./static.generated');
+
+export interface IUserSuppliedRegionInfo {
+  /**
+   * The name of the AWS region, such as `us-east-1`.
+   */
+  readonly name: string;
+
+  /**
+   * The partition in which the region is located (e.g: `aws`, `aws-cn`, ...).
+   *
+   * @default aws
+   */
+  readonly partition?: string;
+
+  /**
+   * The name of the endpoint used for S3 Static Website hosting in the region.
+   *
+   * @default ''
+   */
+  readonly s3WebsiteEndpoint?: string;
+
+  /**
+   * A map from service name (e.g: `s3`, `sqs`, `application-autoscaling`, ...) to the service principal name for that
+   * service in the region (e.g: `codedeploy.<region>.amazonaws.com`).
+   *
+   * @default {}
+   */
+  readonly servicePrincipals?: { [service: string]: string };
+
+  /**
+   * Whether the `AWS::CDK::Metadata` resource is present in the region or not.
+   *
+   * @default false
+   */
+  readonly cdkMetadataResourcePresent?: boolean;
+}
+
+// tslint:disable-next-line:no-var-requires
+const schema: jsonschema.Schema = require('../schema/region-info.schema.json');
+
+export function loadUserData(fromPath: string): { [name: string]: IRegionInfo } {
+  if (!fs.existsSync(fromPath)) {
+    return {};
+  }
+  let userDataArray = JSON.parse(fs.readFileSync(fromPath, { encoding: 'utf8' }));
+  if (!Array.isArray(userDataArray)) {
+    userDataArray = [userDataArray];
+  }
+  const validator = new jsonschema.Validator();
```

Comment:
Isn't `jsonschema` a `devDependency`?

Reply:
There might be a little "oopsie" here :)